### ISSUE TITLE
Shorthand with init not recognized as Pattern when nested in dynamic property

### ISSUE THREAD

**Issue by donnagf** [pvdz/zeparser3#21]
It pass if you try this on AST Explorer

```ts
function ute(nb = ((v= (9869))), {"ad": [r, , , p, ...[, d, , , ...{}]]}, ...{l, [(l --)]: {e = ((new.target)[(((new.target)))]), w = (((function (i, k, x, y) {
}))), "a": xx = (a)}}) {
  for (let q in (((-([, , , s = ((/^/m)), , ] = (([(null), , ]))))))) eval: var t = ((("b"))), v = ((null)), i = (null), k = (arguments)
}
```
You got a similar issue here. Here you found a struct that must be destructured but was not.

```ts
do throw null
 while (({mxpiqdl, "": [m = arguments, a, {}, b = (5.240640260245877e245), (2e308)[""]], "bar": 2e308.constructor = arguments} = (!(class {}))));
export var {"x": g = ((2.0295645404211272e129)), t = (1.0631504437280226e241)(), a} = (((++b))) << this, doe = (i = arguments) => {
}, [] = ((delete "foo")(...false, ...((2e308))) ? this : new null.throw), {p = -((.14))} = /(?:)/muy;
```

**Comment by pvdz** [comment]
Minimal test case: `function f({[x]: {w = x}}) {}`, does not fail when the property is not dynamic so should not be too hard to fix.

**Comment by pvdz** [comment]
Fixed in 3a3ae4b0d40ada7b23ae5fe664de68dd0a631040
